File: backend/src/main/java/com/example/Flicktionary/domain/series/dto/SeriesSummaryResponse.java
package com.example.Flicktionary.domain.series.dto;

import com.example.Flicktionary.domain.series.entity.Series;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;

@AllArgsConstructor
@Builder
@Getter
public class SeriesSummaryResponse {

    private Long id;

    private Long tmdbId;

    private String title;

    private String posterPath;

    private double averageRating;

    private int ratingCount;

    public SeriesSummaryResponse(Series series) {
        this.id = series.getId();
        this.tmdbId = series.getTmdbId();
        this.title = series.getTitle();
        this.posterPath = series.getImageUrl();
        this.averageRating = series.getAverageRating();
        this.ratingCount = series.getRatingCount();
    }
}

File: backend/src/main/kotlin/com/example/Flicktionary/domain/series/dto/SeriesSummaryResponse.kt
package com.example.Flicktionary.domain.series.dto

import com.example.Flicktionary.domain.series.entity.Series

class SeriesSummaryResponse(
    val id: Long,
    val title: String,
    val posterPath: String?,
    val averageRating: Double,
    val ratingCount: Int
) {
    constructor(series: Series) : this(
        series.id,
        series.title,
        series.posterPath,
        series.averageRating,
        series.ratingCount
    )
}
